### PR TITLE
  🏗️ Refactors pytest-simcore plugin fixtures and system tests

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

General cleanup and improvements on
- ``pytest-simcore`` plugin
- ``tests/swarm-deploy`` system tests: reactivates restart service tests
   

## Related issue/s

These changes were necessary while developing a separate PR but are presented here, on its own PR, to facilitate the peer review.

## How to test

all tests are affected

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -511,7 +511,8 @@ known-third-party=enchant
 [DESIGN]
 
 # Maximum number of arguments for function / method
-max-args=7
+# NOTE: due to dependency injection, this number can be quite high
+max-args=10
```
Comment (GitHK) [inline .pylintrc:515]: 👍

File: packages/pytest-simcore/src/pytest_simcore/environment_configs.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def env_devel_dict(env_devel_file: Path) -> Dict[str, Union[str, None]]:
 
 
 @pytest.fixture(scope="function")
-def mock_env_devel_environment(env_devel_dict, monkeypatch):
+def mock_env_testing_environ_varsment(env_devel_dict, monkeypatch):
```
Comment (GitHK) [inline packages/pytest-simcore/src/pytest_simcore/environment_configs.py:23]: `varsment`? what dose that mean?

**Review by GitHK — COMMENTED** [review]
Looks nice, just a comment below.

**Review by GitHK — APPROVED** [review]

File: packages/pytest-simcore/src/pytest_simcore/docker_compose.py [hunk 3]
```diff
@@ -205,27 +227,22 @@ def ops_docker_compose_file(
     return docker_compose_path
 
 
+@pytest.hookimpl()
+def pytest_exception_interact(node, call, report):
+    # get the node root dir (guaranteed to exist)
+    root_directory: Path = Path(node.config.rootdir)
+    failed_test_directory = root_directory / "test_failures" / node.name
+    save_docker_infos(failed_test_directory)
```
Comment (sanderegg) [inline packages/pytest-simcore/src/pytest_simcore/docker_compose.py:235]: that is nice! so this hook gets called whenever a test fails?

**Review by sanderegg — APPROVED** [review]
:+1:

File: packages/pytest-simcore/src/pytest_simcore/environment_configs.py (hunk 2 above)
Comment (pcrespov) [reply packages/pytest-simcore/src/pytest_simcore/environment_configs.py:23]: typo. Corrected.

File: packages/pytest-simcore/src/pytest_simcore/docker_compose.py (hunk 3 above)
Comment (pcrespov) [reply packages/pytest-simcore/src/pytest_simcore/docker_compose.py:235]: @sanderegg , i think you added that some 9 months ago :-)

**Pull request merged**